### PR TITLE
Zoom Pulse Mode

### PR DESCRIPTION
**Pull request by mbebenita**
Depends on #38 

This PR adds a Zoom Pulse button to slowly zoom in and out the Text and SVG demo. The goal is to demonstrate how smooth rendering is.

image

### CODE REVIEW

**Comment by mbebenita** [comment]
I should mention that I couldn't find a good place to clear the `setInterval`, so the timer will keep going after the demo is unmounted. I'm not familiar with this UI toolkit, ... so ideas welcome.

**Comment by trishume** [comment]
This has the risk of the zooming looking choppy even if the underlying rendering isn't, because of the timer being de-synchronized from the frames. Probably better to use `requestAnimationFrame` for the animation, this may also solve your issue with unregistering the timer.

File: demo/client/src/view.ts [hunk 1]
```diff
@@ -75,6 +77,22 @@ export abstract class PathfinderView {
...
 
+    zoomPulse(): void {
+        if (this.pulseTimer) {
+            window.clearInterval(this.pulseTimer);
+            this.pulseTimer = 0;
+            return;
+        }
+        let c = 0;
+        let d = 0.005;
+        this.pulseTimer = window.setInterval(() => {
```
Comment (pcwalton) [inline demo/client/src/view.ts]: Please use `requestAnimationFrame()`

File: demo/client/src/camera.ts [hunk 2]
```diff
@@ -306,6 +311,10 @@ export class PerspectiveCamera extends Camera {
         ]);
     }
 
+    zoom(scale: number): void {
```
Comment (pcwalton) [inline demo/client/src/camera.ts]: Why this change?

Comment (mbebenita) [reply demo/client/src/camera.ts]: I wanted to keep the old `zoom` function private, but I needed something public that only took the `scale` argument. So I renamed the private function to `_zoom` and exposed the new function as `zoom`.

File: demo/client/src/view.ts (hunk 1 above)
Comment (mbebenita) [reply demo/client/src/view.ts]: Will do.

**Comment by mbebenita** [comment]
@trishume good idea, I switched to RAFs in the latest patch.

**Comment by pcwalton** [comment]
Works for me.
